controllers/storagecluster/cephcluster.go

```diff
@@ -605,13 +605,11 @@ func getNetworkSpec(sc ocsv1.StorageCluster) rookCephv1.NetworkSpec {
 	// respect both the old way and the new way for enabling HostNetwork
 	networkSpec.HostNetwork = networkSpec.HostNetwork || sc.Spec.HostNetwork
 
-	// If it's not an external and not a provider cluster always require msgr2
-	if !sc.Spec.AllowRemoteStorageConsumers && !sc.Spec.ExternalStorage.Enable {
-		if networkSpec.Connections == nil {
-			networkSpec.Connections = &rookCephv1.ConnectionsSpec{}
-		}
-		networkSpec.Connections.RequireMsgr2 = true
+	if networkSpec.Connections == nil {
+		networkSpec.Connections = &rookCephv1.ConnectionsSpec{}
 	}
+	// Always require msgr2 for all modes of ODF
+	networkSpec.Connections.RequireMsgr2 = true
```

Comment:
v2 for external mode is controlled by 
`--v2-port-enable` flag, we should not explicitly set here,
Or if there is a need to always make it true we should first do at python script.


Current flow is, 
1) User select v2 port in the python script
2) Script extracts v2 mon ports
3) Internally in rook if  there is v2 detected it will turn cluster:connection to true 
https://github.com/rook/rook/blob/58a0c834bf5bce6782de5c387f3dac732393ef33/pkg/operator/ceph/cluster/cluster_external.go#L126

Reply:
@parth-gr is there any reason why v2 mon ports are not extracted by default if they are available?
Also please look at this BZ here https://bugzilla.redhat.com/show_bug.cgi?id=2180921#c2, This cam for external mode when we tried to use msgrv2 at that point.

Reply:
I don't think we can do that in rook upstream. Is there any hard requirement on the rook version & the Ceph provider clusters's ceph versions? If not upstream user might be running some old version of ceph release. 
@travisn wdyt?

Reply:
I think the behavior we want is:
- When we run the script downstream on the RHCS cluster, if v2 port is available, always capture the v2 port. If no v2 port is available then return the v1 port.
   - In upstream, we still want to default to the v1 port since we don't know if the clients have a recent kernel. 
   - The difference between upstream/downstream hopefully can be automated, so users don't need to be aware of a new flag that they might forget to use.
- If we import the v2 port, then OCS operator can set the `requireMsgr2: true`. If it's still possible downstream to import the v1 port, then we can't set that. But if we can confirm it's not supported for RHCS, then we could always assume the v2 port is required for downstream.

Reply:
@parth-gr @travisn  do we really need a separate flag to extract the v2 ports?
I noticed the mons have the same address for both v1 & v2, so if we have the address of the mon for v1, for v2 shouldn't it be simply done by replacing port 6789 with port 3300?

```
sh-5.1$ ceph mon dump
epoch 4
fsid 053e09a7-bf58-4312-a3b9-2d05b5723fbf
last_changed 2023-04-30T20:56:08.009439+0000
created 2023-04-30T20:37:36.381975+0000
min_mon_release 17 (quincy)
election_strategy: 1
0: [v2:172.30.15.98:3300/0,v1:172.30.15.98:6789/0] mon.a
1: [v2:172.30.230.232:3300/0,v1:172.30.230.232:6789/0] mon.b
2: [v2:172.30.44.179:3300/0,v1:172.30.44.179:6789/0] mon.c
dumped monmap epoch 4
```

Reply:
@malayparida2000 we need a separate flag to identify what user is requesting.
Till now in standalone cluster there can 4 conditions,
> mons only have v1 port
> mons only have v2 port
> mons have v1 and v2 port
> custom mons from user(not supported but there a BZ opened)

Reply:
Hi, on latest RHCS (7.1) by default mon listen to both v1 (6789) and v2 (3300) ports, but, as previously stated, a customer can have very old clients and/or decide (even if not recommended) to enable v1 only.
So, for external cluster, I agree with @travisn approach

Reply:
iirc. v2+v1 notation were introduced when msgrv2 was introduced and it's still the default today `[v2:mon1:3300/0,v1:mon1:6789/0]`

This should allow us to just probe for v2, against each mon, and, cluster-wide, fall-back to v1, as long as features from v2 are not wished / required.